File: backend/app/graphql/types/income.py
from __future__ import annotations

import enum

import strawberry
from datetime import date, datetime
from decimal import Decimal


@strawberry.enum
class IncomeTypeEnum(enum.Enum):
    SALARY = "salary"
    FREELANCE = "freelance"
    INVESTMENTS = "investments"
    RENTAL = "rental"
    BUSINESS = "business"
    OTHER = "other"


@strawberry.type
class IncomeType:
    id: strawberry.ID
    name: str
    amount: Decimal
    income_type: str
    is_active: bool
    start_date: date | None
    notes: str | None
    currency: str
    is_gross: bool
    tax_rate: Decimal | None
    other_fees: Decimal | None
    net_amount: Decimal
    created_at: datetime
    updated_at: datetime


@strawberry.type
class IncomeConnection:
    items: list[IncomeType]
    total_count: int
    has_more: bool

File: frontend/src/graphql/queries/income.ts
import { gql } from "@apollo/client";

export const GET_INCOMES = gql`
  query GetIncomes($isActive: Boolean, $limit: Int = 50, $offset: Int = 0) {
    incomes(isActive: $isActive, limit: $limit, offset: $offset) {
      items {
        id
        name
        amount
        incomeType
        isActive
        startDate
        notes
        currency
        isGross
        taxRate
        otherFees
        netAmount
        createdAt
        updatedAt
      }
      totalCount
      hasMore
    }
  }
`;

export const GET_INCOME = gql`
  query GetIncome($id: ID!) {
    income(id: $id) {
      id
      name
      amount
      incomeType
      isActive
      startDate
      notes
      currency
      isGross
      taxRate
      otherFees
      netAmount
      createdAt
      updatedAt
    }
  }
`;
